packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -555,24 +556,32 @@ const ProductList = (props) => {
                             itemsBefore={
                                 category?.categories
                                     ? [
-                                          <CategoryLinks key="itemsBefore" category={category} />,
-                                          <StoreInventoryFilter
-                                              key="storeInventoryFilter"
-                                              toggleFilter={toggleFilter}
-                                              selectedFilters={
-                                                  productSearchResult?.selectedRefinements || {}
-                                              }
-                                          />
-                                      ]
+                                          <CategoryLinks
+                                              key="itemsBefore"
+                                              category={category}
+                                              onSelect={onClose}
+                                          />,
+                                          STORE_LOCATOR_IS_ENABLED && (
+                                              <StoreInventoryFilter
+                                                  key="storeInventoryFilter"
+                                                  toggleFilter={toggleFilter}
+                                                  selectedFilters={
+                                                      productSearchResult?.selectedRefinements || {}
+                                                  }
+                                              />
+                                          )
+                                      ].filter(Boolean)
                                     : [
-                                          <StoreInventoryFilter
-                                              key="storeInventoryFilter"
-                                              toggleFilter={toggleFilter}
-                                              selectedFilters={
-                                                  productSearchResult?.selectedRefinements || {}
-                                              }
-                                          />
-                                      ]
+                                          STORE_LOCATOR_IS_ENABLED && (
+                                              <StoreInventoryFilter
+                                                  key="storeInventoryFilter"
+                                                  toggleFilter={toggleFilter}
+                                                  selectedFilters={
+                                                      productSearchResult?.selectedRefinements || {}
+                                                  }
+                                              />
+                                          )
+                                      ].filter(Boolean)
```

Comment:
How about we keep this block DRY by creating a variable to keep the component to be rendered? 10+ lines for a prop seems too much for readablility